src/Controller/Backend/UserEditController.php

```diff
@@ -43,44 +43,117 @@ class UserEditController extends TwigAwareController implements BackendZoneInter
     /** @var EventDispatcherInterface */
     private $dispatcher;
 
+    protected $defaultLocale;
+
     public function __construct(
         UrlGeneratorInterface $urlGenerator,
         EntityManagerInterface $em,
         UserPasswordEncoderInterface $passwordEncoder,
         CsrfTokenManagerInterface $csrfTokenManager,
-        EventDispatcherInterface $dispatcher
+        EventDispatcherInterface $dispatcher,
+        string $defaultLocale
     ) {
         $this->urlGenerator = $urlGenerator;
         $this->em = $em;
         $this->passwordEncoder = $passwordEncoder;
         $this->csrfTokenManager = $csrfTokenManager;
         $this->dispatcher = $dispatcher;
+        $this->defaultLocale = $defaultLocale;
     }
 
     /**
-     * @Route("/user-edit/{id}", methods={"GET"}, name="bolt_user_edit", requirements={"id": "\d+"})
+     * @Route("/user-edit/add", methods={"GET","POST"}, name="bolt_user_add")
      */
-    public function edit(?User $user): Response
+    public function add(Request $request): Response
     {
-        if (! $user instanceof User) {
-            $user = UserRepository::factory();
-            $suggestedPassword = Str::generatePassword();
-        } else {
-            $suggestedPassword = '';
+        $user = UserRepository::factory();
+        $submitted_data = $request->get('user_edit');
+
+        $event = new UserEvent($user);
+        $this->dispatcher->dispatch($event, UserEvent::ON_ADD);
```

Comment:
Great idea to add more user events!

Wonder if, for the sake of consistency with `ContentEvent`, we should have `PRE_SAVE` and `POST_SAVE` somewhere? Just a thought, maybe it's out-of-scope for this PR.

Reply:
Love that Idea, will add that right now.
Since add and edit both use the private function `_handleValidFormSubmit` I will put them in there.

Reply:
Awesome 😎